var/spack/repos/builtin/packages/trilinos/package.py

```diff
@@ -418,18 +420,15 @@ class Trilinos(CMakePackage, CudaPackage, ROCmPackage):
         depends_on("kokkos-kernels@4.2.01", when="@15.1:15")
         depends_on("kokkos-kernels@4.1.00", when="@14.4:15.0")
 
-    depends_on("kokkos +wrapper", when="trilinos@14.4.0: +kokkos +wrapper")
-    depends_on("kokkos ~wrapper", when="trilinos@14.4.0: +kokkos ~wrapper")
-
     for a in CudaPackage.cuda_arch_values:
-        arch_str = "+cuda cuda_arch={0}".format(a)
-        kokkos_spec = "kokkos {0}".format(arch_str)
-        depends_on(kokkos_spec, when="@14.4.0: +kokkos {0}".format(arch_str))
+        arch_str = f"+cuda cuda_arch={a}"
+        kokkos_spec = f"kokkos {arch_str}"
+        depends_on(kokkos_spec, when=f"@14.4: +kokkos {arch_str}")
```

Comment:
also one more level of indent and drop the `@14.4: +kokkos` bit?

Reply:
Argh, I did not even check what this block was doing. 
Now I've shortened it vertically and moved to the common `with when` block.